src/storage/chaindb.md

```diff
@@ -0,0 +1,104 @@
+# The ChainDB format
+
+In the Haskell reference implementation, the Storage layer manages a `ChainDB`,
+whose storage components are the `ImmutableDB`, `VolatileDB` and Ledger State
+snapshots:
+
+| Component              | Responsibility                                                                                                                                          |
+|:-----------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------|
+| ImmutableDB            | Store definitive blocks and headers sequentially                                                                                                        |
+| VolatileDB             | Store a bag of non-definitive blocks and headers. In particular it contains the blocks which, when linked sequentially, form the current selected chain |
+| Ledger State snapshots | Periodically store the ledger state at the tip of the ImmutableDB                                                                                       |
```

Comment:
Do these refer to components of the database? If so perhaps we should find another name to distinguish them from the Consensus (software) components.

Reply:
Naming them "Immutable/Volatile Database" now